### PR TITLE
Remove mutex in message sender

### PR DESCRIPTION
**Pull request by LeoHChen**
atomic.Value is faster than the Mutex we used in messageSender.

### CODE REVIEW

**Comment by LeoHChen** [comment]
> According to my knowledge, `atomic.Value` is only used for structure. Do you think using `atomic.StoreUint64` and `atomic.StoreBool` would be better?
> 
> If the naming without atomic keyword seems confusing, do you think it is a better practice to use a wrapped atomic module? ([uber go style guide](https://github.com/uber-go/guide/blob/master/style.md#use-gouberorgatomic)).

sounds good, we can use explicit func to store/load value. A wrapper module is a bit overkill now.

**Comment by JackyWYX** [comment]
> > According to my knowledge, `atomic.Value` is only used for structure. Do you think using `atomic.StoreUint64` and `atomic.StoreBool` would be better?
> > If the naming without atomic keyword seems confusing, do you think it is a better practice to use a wrapped atomic module? ([uber go style guide](https://github.com/uber-go/guide/blob/master/style.md#use-gouberorgatomic)).
> 
> sounds good, we can use explicit func to store/load value. A wrapper module is a bit overkill now.

Cannot agree more. Explicit call of load have much better performance

File: consensus/consensus_service.go [hunk 1]
```diff
@@ -273,9 +274,7 @@ func (consensus *Consensus) checkViewID(msg *FBFTMessage) error {
 
 // SetBlockNum sets the blockNum in consensus object, called at node bootstrap
 func (consensus *Consensus) SetBlockNum(blockNum uint64) {
```
Comment (JackyWYX) [inline consensus/consensus_service.go:276]: Do you think there is a race condition at `consensus/consensus_v2.go:263`.

Comment (LeoHChen) [reply consensus/consensus_service.go:276]: Agree. there could be race condition in all the reading.  Our original code wasn't protecting all the read operation at all.  I'd do it later.  This PR is getting too big now.

Comment (LeoHChen) [reply consensus/consensus_service.go:276]: updated with atomic.AddUint64

**Comment by JackyWYX** [comment]
More read protection can be done later. Look good to me.

**Pull request merged**